File: hi.rs
#[allow(unused_variables)]
fn main() {
    for _ in 0..98 {
        let _ = String::from("eat pie");
    }

    let mut x = String::new();
    for _ in 0..1_000_000 {
        x.push('a');
    }

    let squiggly = f32::sqrt(5.0);
    println!("Hi");
}

File: hi.hs
module Main where

import Data.Foldable (forM_)

main :: IO ()
main = do
  forM_ [1..98] (let eat_pie = "eat pie" in \ _ -> return eat_pie)
  let x = iterate ('a' :) [] !! 1000000
  let squiggly = sqrt 5
  putStr "hi\n"
